### PR TITLE
CC-26461 Backporting improvements to the password reset workflow

### PR DESCRIPTION
**Pull request by floriankraemer**
Branch: backport/5.51.6
Ticket: https://spryker.atlassian.net/browse/CC-26461
Version: 5.51.6
- Strategy: minor

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Customer                | patch                 |                       |

-----------------------------------------

##### Change log

Improvements

- Adjusted `CustomerFacade::restorePassword()` with the behavior that customer password resets expire after two hours.
- Introduced `CustomerConfig::isCustomerPasswordResetExpirationEnabled(): bool()` to enable or disable the reset expiration. It is enabled by default.
- Introduced `CustomerConfig::getCustomerPasswordResetExpirationPeriod()` to get a DateTime-compatible string to define the expiration threshold. The default is two hours.

### CODE REVIEW

File: src/Spryker/Zed/Customer/Business/Customer/Customer.php [hunk 1]
```diff
@@ -505,6 +508,14 @@ public function restorePassword(CustomerTransfer $customerTransfer)
             return $customerResponseTransfer;
         }
 
+        $customerResponseTransfer = $this
```
Comment (dmiseev) [inline src/Spryker/Zed/Customer/Business/Customer/Customer.php:511]: the test file should be also copied from the original PR (`Bundles/Customer/tests/SprykerTest/Zed/Customer/Business/Facade/RestorePasswordTest.php`).

**Comment by dmiseev** [comment]
Module name in the changelogs is wrong (not a `Kernel`).

File: src/Spryker/Zed/Customer/Business/Customer/Customer.php (hunk 1 above)
Comment (floriankraemer) [reply src/Spryker/Zed/Customer/Business/Customer/Customer.php:511]: I thought we don't backport tests? 😕

Comment (dmiseev) [reply src/Spryker/Zed/Customer/Business/Customer/Customer.php:511]: we backporting all diffs

**Pull request closed without merging**